driver/src/test/functional/com/mongodb/reactivestreams/client/gridfs/GridFSPublisherSpecification.groovy

```diff
@@ -126,7 +127,36 @@ class GridFSPublisherSpecification extends FunctionalSpecification {
         def data = runAndCollect(gridFSBucket.&downloadToPublisher, fileId)
 
         then:
-        concatByteBuffers(data) == concatByteBuffers([ByteBuffer.wrap(contentBytes), ByteBuffer.wrap(contentBytes)])
+        concatByteBuffers(data) == contentBytes
+    }
+
+    def 'should respect the outer subscription request amount'() {
+        given:
+        def contentBytes = multiChunkString.getBytes()
+        def options = new GridFSUploadOptions().chunkSizeBytes(contentBytes.length)
+
+        when:
+        def publisher = gridFSBucket.uploadFromPublisher('myFile',
+        createPublisher(ByteBuffer.wrap(contentBytes), ByteBuffer.wrap(contentBytes),
+                ByteBuffer.wrap(contentBytes)), options)
+
+        def subscriber = new ObservableSubscriber()
+        publisher.subscribe(subscriber)
+        def fileId = subscriber.await(1, 60, SECONDS).getReceived().get(0)
```

Comment:
This ensures that we only request 1 fileId and tests that internally all the source Publisher is consumed.